.npmrc

```diff
@@ -0,0 +1 @@
+save-exact=true
```

Comment:
.npmrc is **the** npm configuration file. Avoiding pinning of versions to avoid using the configuration file makes no sense. What are we going to do if we want to configure something else? We shouldn’t avoid using the configuration file because a developer _can_ put credentials in them.

> At this point it feels kinda like exact pinning is causing more problems than it solves.

I principally disagree with this statement.

Reply:
I think the answer to that is that if you use the recommended defaults that the tool comes with you then don't have to configure things, and things just tend to work with no issues.

If we need to configure things differently to the vast majority of the JS/TS ecosystem we should be asking ourselves why.

Anyway no action needed here